groups/views.py

```diff
@@ -135,8 +156,39 @@ def get_context_data(self, *args, **kwargs):
     def form_valid(self, form):
         form.instance.user = self.request.user
         form.instance.discussion = self.discussion
+        self.email_subscribers(form.instance)
         return super(CommentPostView, self).form_valid(form)
 
+    def users_to_notify(self, comment):
+        """
+        Return subscribers to the comment's discussion or its parent group.
+
+        Exclude anyone who's explicitly ignored this discussion, and the person who
+        posted the comment.
+        """
+        discussion = comment.discussion
+        discussion_subscribers = discussion.subscribers.all()
+        group_subscribers = discussion.group.watchers.exclude(
+            ignored_discussions=discussion
+        )
+
+        all_subscribers = discussion_subscribers | group_subscribers
+        return all_subscribers.exclude(pk=comment.user.pk)
```

Comment:
Is there a functional benefit besides some small test tweaking?  I'm not sure it's worth it, especially once you're adding proxy models.  I could make it a static method?